projects/laji/src/app/shared-modules/select/select-subcategories/select-subcategories.component.ts

```diff
@@ -0,0 +1,607 @@
+        }
+        return;
+      case 'ArrowDown':
+        if (this.selectedIdx[category] >= this.unselectedOptions[category].length - 1) {
+          this.selectedIdx[category] = 0;
+        } else {
+          this.selectedIdx[category]++;
+        }
+        return;
+      default:
+        this.selectedIdx[category] = -1;
+    }
+    this.filterInput.next(value);
+  }
+
+  track(idx, item) {
+    return item.id;
+  }
+
+  private initOptions(selected) {
+    if (!this.options || !this.selected) {
+      return;
+    }
+    // this.selectedOptions = [];
+    if (!selected) {
+      this.unselectedOptions = [];
+      return;
+    }
+
+  for (const i in this.selected) {
+    this.selectedOptions[i] = [];
+    this.unselectedOptions[i] = [];
+    const countObj = this.selected[i].filter(item => item.checkboxValue === true).length;
+    this.options[i].map(option => {
+      if (countObj > 0) {
+        selected[i].map(item => {
+          if (item.id === option.id) {
+            this.selectedOptions[i].push({'id': option.id, 'value': option.value, 'info': option.info, 'checkboxValue': item.checkboxValue});
+          } else {
+            this.unselectedOptions[i].push({'id': option.id, 'value': option.value, 'info': option.info, 'checkboxValue': item.checkboxValue});
+          }
+        });
+      } else {
+        if (selected[i].includes(option.id)) {
+          if (!this.selectedOptions[i]) {
+            this.selectedOptions[i] = option.id;
+          } else {
+            this.selectedOptions[i].push(option.id);
+          }
+        } else {
+          if (!this.unselectedOptions[i]) {
+            this.unselectedOptions[i] = option.id;
+          } else {
+            this.unselectedOptions[i].push(option.id);
+          }
+        }
+      }
+    });
+  }
+
+  }
+
+
+  private checkSubcategoriesExceptGlobalAreEquals(selected) {
+    const keys = Object.keys(selected);
+    const filteredKeys = keys.filter(item => item !== 'GLOBAL');
+
+    if (filteredKeys.length < this.subCategories.length - 1) {
+      return false;
+    }
+
+    let countEmpty = 0;
+    for (let i = 0; i < filteredKeys.length; i++) {
+      if (selected[filteredKeys[i]].length === 0 || selected[filteredKeys[i]] === undefined) {
+        countEmpty++;
+      }
+    }
+
+    if (countEmpty === filteredKeys.length) {
+      this.selected['GLOBAL'].map(x => x.checkboxValue = undefined);
+      return false;
+    }
+
+    for (let i = 0; i < filteredKeys.length - 1; i++) {
+        if (!this.arrayEquals(selected[filteredKeys[i]], selected[filteredKeys[i + 1]])) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+
+  private arrayEquals(a, b) {
+    return Array.isArray(a) &&
+      Array.isArray(b) &&
+      a.length === b.length &&
+      a.every((val) => b.indexOf(val) > -1);
+  }
+
+  private buildSelectedOptions(filters) {
+    const tmpOptions = this.options;
+    this.activatedRoute.queryParams.subscribe(params => {
```

Comment:
There is no unsubscribe to this subscription and this is subscribed on every on change in this component causing this to fire multiple times when toggling any of the components values for example. This is also reading query parameters directly from route instead of using query input. Is there a reason why query input could not be used instead?